lib/seahorn/Houdini.cc

```diff
@@ -0,0 +1,292 @@
+#include "seahorn/Houdini.hh"
+#include "seahorn/HornifyModule.hh"
+#include "seahorn/HornClauseDBTransf.hh"
+#include "seahorn/HornClauseDB.hh"
+
+#include "llvm/IR/Function.h"
+#include "llvm/Support/CommandLine.h"
+
+#include "ufo/Expr.hpp"
+#include "ufo/Smt/Z3n.hpp"
+#include "ufo/Smt/EZ3.hh"
+#include <vector>
+
+using namespace llvm;
+
+namespace seahorn
+{
+  char Houdini::ID = 0;
+
+  bool Houdini::runOnModule (Module &M)
+  {
+    HornifyModule &hm = getAnalysis<HornifyModule> ();
+
+    //load the Horn clause database
+    auto &db = hm.getHornClauseDB ();
+    db.buildIndexes ();
+
+    workListAlgo(db, hm);
+    //getUseRuleSet(db);
+
+    return false;
+  }
+
+  void Houdini::getAnalysisUsage (AnalysisUsage &AU) const
+  {
+    AU.addRequired<HornifyModule> ();
+    AU.setPreservesAll();
+  }
+
+  void Houdini::printDB (const HornClauseDB &db)
+  {
+    outs () << db << "\n";
+  }
+
+  void Houdini::printHello ()
+  {
+    outs () << "Hello there.\n";
+  }
+
+  void Houdini::getUseRuleSet(HornClauseDB &db)
+  {
+	  auto &workList = db.getRules();
```

Comment:
this seems wrong. The worklist should not be a reference to the rules in the database.